src/function/scalar/operators/arithmetic.cpp

```diff
@@ -221,7 +221,7 @@ unique_ptr<FunctionData> BindDecimalAddSubtract(ClientContext &context, ScalarFu
 		// first check if the cast is necessary
 		// if the argument has a matching scale and internal type as the output type, no casting is necessary
 		auto &argument_type = arguments[i]->return_type;
-		uint8_t width, scale;
+		uint8_t width, scale = 255u;
```

Comment:
Named value would be better, unsure what's clearly out-of-bound as scale.

Reply:
Perhaps always initialize it in `GetDecimalProperties`?

This is also rather confusing because only `scale` is initialized which I suppose is also not the intention?

Reply:
I think in this case actually an `InternalException` should be thrown if `GetDecimalProperties` returns false

Reply:
For now I initialised both scale and width to 255u inside GetDecimalProperties, throw is probably better, but requires changing interface (basically return value has no meaning anymore), will think about that later.